transcom/mymove
# INT B-20080 Incorrect GBLOC in Search Results

## [B-20080](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20080)

## Summary
> [!TIP]
> This is a lot of detail for a small change but the scenario is odd and took me a while to understand. Plus I've been wanting to use this new markdown feature.

A move's origin GBLOC depends on the viewer and the move's status. For the SC, it is based on the origin duty location; for other office users, it's based on the pickup address. Before counseling, a move search should use the origin duty location, and after counseling, it should use the pickup address. These two zips usually fall within the same GBLOC, but in the rare case they differ the move search results were always using the origin duty location regardless of the move's status.

The ticket's description suggests the logic for selecting the correct address to base the GBLOC on was wrong, however, the logic appears to be fine. Instead it was that the pickup address GBLOCs were not included in the search query so the array being checked (`move.ShipmentGBLOC[]`) was always empty so it was falling back on the origin duty location. This updates the search query to return the necessary data.

Is there anything you would like reviewers to give additional scrutiny?
If you have any ideas of how to add tests for this I'm open to them. This seemed niche enough to not matter too much.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the customer app as a new or existing customer.
2. Create a move noting the GBLOC of your selected duty location. I used Scott AFB, AGFM.
3. Add a new shipment (PPM or HHG but ideally HHG so you can easily access it as a TOO)
4. Select a pickup location that is in a different GBLOC than your selected duty location. (I used Savannah GA 31406, CNNQ)
5. Submit the move as a customer.
6. Log in as a services counselor for the GBLOC of the above move's Duty Location.
8. Verify the move is in your queue.
9. Search for the move from the move searcher and verify the origin GBLOC displayed is that of the origin duty location.
10. As the counselor approve the move.
11. Verify the move is no longer in your counseling queue.
12. Search for the move again and verify the listed origin GBLOC is now that of the pickup address.
13. Log out as the counselor.
14. Log in as a TOO in the GBLOC of the shipment's the pickup address.
15. Verify the move is in your queue.
16. Search for the move from the move searcher and verify the origin GBLOC displayed is that of the shipment's pickup location.

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
